<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.26">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="大番薯本薯">
<meta name="dcterms.date" content="2025-12-07">
<title>Nextclade</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-587c61ba64f3a5504c4d52d930310e48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-6fb6fcbaca076a07ad0b43b7bab862d0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "Search",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script src="../site_libs/quarto-diagram/mermaid.min.js"></script><script src="../site_libs/quarto-diagram/mermaid-init.js"></script><link href="../site_libs/quarto-diagram/mermaid.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../styles.css">
</head>
<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">RSSPtho</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-books" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Books</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-books">
<li>
    <a class="dropdown-item" href="../Books/ggplot2/index.html">
 <span class="dropdown-text">ggplot2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Books/R4DS2/index.html">
 <span class="dropdown-text">R for Data Science(2e)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Books/Tidy Modeling with R/index.html">
 <span class="dropdown-text">Tidy Modeling with R</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Books/Advanced R(2e)/index.html">
 <span class="dropdown-text">Advanced R(2e)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Books/quarto/Project Basics.html">
 <span class="dropdown-text">Quarto</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../Books/Js4R/00 Preface.qmd">
 <span class="dropdown-text">Js4R</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item">
    <a class="nav-link" href="../blog.html"> 
<span class="menu-text">Blog</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About me</span></a>
  </li>  
</ul>
<div class="quarto-navbar-tools">
    <a href="https://github.com/hanguojun007/Blog" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#nextstrain-%E4%B8%8E-nextclade" id="toc-nextstrain-与-nextclade" class="nav-link active" data-scroll-target="#nextstrain-%E4%B8%8E-nextclade">Nextstrain 与 Nextclade</a>
  <ul class="collapse">
<li><a href="#augur-%E5%92%8C-auspice" id="toc-augur-和-auspice" class="nav-link" data-scroll-target="#augur-%E5%92%8C-auspice">Augur 和 Auspice</a></li>
  <li><a href="#nextclade-%E5%B7%A5%E4%BD%9C%E6%B5%81%E7%A8%8B" id="toc-nextclade-工作流程" class="nav-link" data-scroll-target="#nextclade-%E5%B7%A5%E4%BD%9C%E6%B5%81%E7%A8%8B">nextclade 工作流程</a></li>
  </ul>
</li>
  <li>
<a href="#nextclade-cli" id="toc-nextclade-cli" class="nav-link" data-scroll-target="#nextclade-cli">Nextclade CLI</a>
  <ul class="collapse">
<li><a href="#%E5%AE%89%E8%A3%85" id="toc-安装" class="nav-link" data-scroll-target="#%E5%AE%89%E8%A3%85">安装</a></li>
  <li><a href="#cli-%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95" id="toc-cli-使用方法" class="nav-link" data-scroll-target="#cli-%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95">CLI 使用方法</a></li>
  </ul>
</li>
  <li>
<a href="#nextclade-%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6" id="toc-nextclade-输入文件" class="nav-link" data-scroll-target="#nextclade-%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6">Nextclade 输入文件</a>
  <ul class="collapse">
<li><a href="#%E6%9F%A5%E8%AF%A2%E5%BA%8F%E5%88%97" id="toc-查询序列" class="nav-link" data-scroll-target="#%E6%9F%A5%E8%AF%A2%E5%BA%8F%E5%88%97">查询序列</a></li>
  <li><a href="#dataset" id="toc-dataset" class="nav-link" data-scroll-target="#dataset">dataset</a></li>
  </ul>
</li>
  <li>
<a href="#nextclade-%E8%BE%93%E5%87%BA%E6%96%87%E4%BB%B6" id="toc-nextclade-输出文件" class="nav-link" data-scroll-target="#nextclade-%E8%BE%93%E5%87%BA%E6%96%87%E4%BB%B6">Nextclade 输出文件</a>
  <ul class="collapse">
<li><a href="#nucleotide-alignment" id="toc-nucleotide-alignment" class="nav-link" data-scroll-target="#nucleotide-alignment">Nucleotide alignment</a></li>
  <li><a href="#translations" id="toc-translations" class="nav-link" data-scroll-target="#translations">Translations</a></li>
  <li><a href="#analysis-results" id="toc-analysis-results" class="nav-link" data-scroll-target="#analysis-results">Analysis results</a></li>
  <li><a href="#phylogenetic-tree" id="toc-phylogenetic-tree" class="nav-link" data-scroll-target="#phylogenetic-tree">Phylogenetic tree</a></li>
  <li><a href="#genome-annotation-of-query-sequences" id="toc-genome-annotation-of-query-sequences" class="nav-link" data-scroll-target="#genome-annotation-of-query-sequences">Genome annotation of query sequences</a></li>
  </ul>
</li>
  <li>
<a href="#algorithm" id="toc-algorithm" class="nav-link" data-scroll-target="#algorithm">Algorithm</a>
  <ul class="collapse">
<li><a href="#sequence-alignment" id="toc-sequence-alignment" class="nav-link" data-scroll-target="#sequence-alignment">Sequence alignment</a></li>
  <li><a href="#translation" id="toc-translation" class="nav-link" data-scroll-target="#translation">Translation</a></li>
  <li><a href="#phylogenetic-placement" id="toc-phylogenetic-placement" class="nav-link" data-scroll-target="#phylogenetic-placement">Phylogenetic placement</a></li>
  <li><a href="#clade-assignment" id="toc-clade-assignment" class="nav-link" data-scroll-target="#clade-assignment">Clade assignment</a></li>
  <li><a href="#mutation-calling" id="toc-mutation-calling" class="nav-link" data-scroll-target="#mutation-calling">Mutation calling</a></li>
  <li><a href="#quality-control-qc" id="toc-quality-control-qc" class="nav-link" data-scroll-target="#quality-control-qc">Quality Control (QC)</a></li>
  <li><a href="#detection-of-pcr-primer-changes" id="toc-detection-of-pcr-primer-changes" class="nav-link" data-scroll-target="#detection-of-pcr-primer-changes">Detection of PCR primer changes</a></li>
  </ul>
</li>
  </ul></nav>
    <div class="quarto-margin-footer"><div class="margin-footer-item">
<p><img src="../logo.jpg" class="conditional-footer img-fluid" width="200"></p>
</div></div></div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<h1 class="title">Nextclade</h1>
  <div class="quarto-categories">
    <div class="quarto-category">nextclade</div>
  </div>
  </div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>大番薯本薯 </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">December 7, 2025</p>
    </div>
  </div>
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">December 7, 2025</p>
    </div>
  </div>
    
  </div>
  


</header><p><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/index.html">官方手册</a></p>
<p>主要分两种方式使用：</p>
<ul>
<li><p><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-web/index.html">Nextclade Web</a></p></li>
<li><p><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-cli/index.html">Nextclade CLI</a></p></li>
</ul>
<section id="nextstrain-与-nextclade" class="level2"><h2 class="anchored" data-anchor-id="nextstrain-与-nextclade">Nextstrain 与 Nextclade</h2>
<p><a href="https://docs.nextstrain.org/">Nextstrain</a> 项目提供了一套对数据进行处理，分析，可视化的开源软件工具，<a href="https://docs.nextstrain.org/projects/nextclade/en/stable/index.html#">Nextclade</a> 的分析基于这些工具，例如：nextclade 使用的输入数据集“dataset”是由 <a href="https://docs.nextstrain.org/projects/augur/en/stable/">Augur</a> 工具生成，输出结果的可视化由 <a href="https://docs.nextstrain.org/projects/auspice/en/stable/">Auspice</a> 工具支持。</p>
<section id="augur-和-auspice" class="level3"><h3 class="anchored" data-anchor-id="augur-和-auspice">Augur 和 Auspice</h3>
<p>Augur 与 Auspice 的关系如下：</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"></figure><p></p>
<div>
<pre class="mermaid mermaid-js">graph LR
  A[/equences.fasta/] --&gt; B(filter)
  C[/metadata.tsv/] --&gt; B
  subgraph Augur
  B --&gt; D(align)
  D --&gt; E(tree)
  E --&gt; F(refine)
  F --&gt; G(export)
  end
  G --&gt; H[(Dataset)]
  H --&gt; I[Auspice]

  classDef input fill:#d3d3d3,stroke:black
  classDef Tool fill:#a6cee3,stroke:#1f78b4
  classDef tool fill:white,stroke:#1f78b4

  class A,C input
  class B,D,E,F,G tool
  class Augur,I Tool
</pre>
</div>
<p></p>
</div>
</div>
</div>
</section><section id="nextclade-工作流程" class="level3"><h3 class="anchored" data-anchor-id="nextclade-工作流程">nextclade 工作流程</h3>
<ul>
<li>将查询序列与参考序列进行比对</li>
<li>检测突变</li>
<li>根据突变进行基于“参考树”的分支</li>
<li>将查询序列整合到“参考树”中</li>
</ul></section></section><section id="nextclade-cli" class="level2"><h2 class="anchored" data-anchor-id="nextclade-cli">Nextclade CLI</h2>
<section id="安装" class="level3"><h3 class="anchored" data-anchor-id="安装">安装</h3>
<p>有四种安装方式，暂时先使用docker镜像，进行初步研究。</p>
<ul>
<li><p><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-cli/installation/standalone.html">Standalone download</a></p></li>
<li><p><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-cli/installation/docker.html">With Docker</a></p></li>
<li><p><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-cli/installation/conda.html">With Conda</a></p></li>
<li><p><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-cli/installation/build-from-source.html">Build from source</a></p></li>
</ul>
<section id="docker" class="level4"><h4 class="anchored" data-anchor-id="docker">docker</h4>
<p>下载方式：</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb1"><pre class="sourceCode sh code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">docker</span> pull nextstrain/nextclade:latest</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-it</span> <span class="at">--rm</span> nextstrain/nextclade:latest nextclade <span class="at">--help</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>使用示例：</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb2"><pre class="sourceCode sh code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-it</span> <span class="at">--rm</span> <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--volume</span><span class="op">=</span><span class="st">"</span><span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span><span class="st">:/data/"</span> <span class="dt">\</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--user</span><span class="op">=</span><span class="st">"</span><span class="va">$(</span><span class="fu">id</span> <span class="at">-u</span><span class="va">)</span><span class="st">:</span><span class="va">$(</span><span class="fu">id</span> <span class="at">-g</span><span class="va">)</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"nextstrain/nextclade"</span> <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  nextclade run <span class="dt">\</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">--dataset-name</span><span class="op">=</span><span class="st">"sars-cov-2"</span> <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">--output-dir</span><span class="op">=</span><span class="st">"/data/output/"</span> <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>      <span class="st">"/data/my_sequences.fasta"</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-it</span> <span class="at">--name</span> nextclade <span class="at">-v</span> /home/hgj/D/NextClade/Test:/root/data nextstrain/nextclade</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</section></section><section id="cli-使用方法" class="level3"><h3 class="anchored" data-anchor-id="cli-使用方法">CLI 使用方法</h3>
<p><code>nextclade [OPTIONS] &lt;COMMAND&gt;</code></p>
<p>主要的command有：</p>
<ul>
<li>
<code>run</code>：运行 Nextclade</li>
<li>
<code>dataset</code>：获取数据集信息</li>
<li>
<code>schema</code>：生成json格式模板</li>
</ul>
<p>详细参数见，<code>nextclade -h</code> 或 <code>nextclade &lt;command&gt; -h</code>，及官方参考<a href="https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-cli/reference.html">refrence</a>。</p>
<p>测试示例：</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb3"><pre class="sourceCode sh code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nextclade</span> run <span class="at">--input-dataset</span> data/sars-cov-2 <span class="at">--output-all</span><span class="op">=</span>output/ data/sars-cov-2/sequences.fasta</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>输入<code>--input-dataset</code>与输出<code>--output-all</code>可以更细致地进行自定义。</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb4"><pre class="sourceCode sh code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nextclade</span> run <span class="dt">\</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>   <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">--include-reference</span> <span class="dt">\</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">--in-order</span> <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>   <span class="co"># 输入数据集控制</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>   <span class="ex">--input-dataset=data/sars-cov-2</span> <span class="dt">\</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>   <span class="at">--input-ref</span><span class="op">=</span>data/sars-cov-2/reference.fasta <span class="dt">\</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>   <span class="at">--input-annotation</span><span class="op">=</span>data/sars-cov-2/genome_annotation.gff3 <span class="dt">\</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>   <span class="at">--cds-selection</span><span class="op">=</span>E,M,N,ORF1a,ORF1b,ORF3a,ORF6,ORF7a,ORF7b,ORF8,ORF9b,S <span class="dt">\</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>   <span class="at">--input-tree</span><span class="op">=</span>data/sars-cov-2/tree.json <span class="dt">\</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>   <span class="at">--input-pathogen-json</span><span class="op">=</span>data/sars-cov-2/pathogen.json <span class="dt">\</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>   <span class="co"># 输出数据控制</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>   <span class="ex">--output-fasta=output/nextclade.aligned.fasta.gz</span> <span class="dt">\</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>   <span class="at">--output-json</span><span class="op">=</span>output/nextclade.json <span class="dt">\</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>   <span class="at">--output-ndjson</span><span class="op">=</span>output/nextclade.ndjson <span class="dt">\</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>   <span class="at">--output-csv</span><span class="op">=</span>output/nextclade.csv <span class="dt">\</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>   <span class="at">--output-tsv</span><span class="op">=</span>output/nextclade.tsv <span class="dt">\</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>   <span class="at">--output-tree</span><span class="op">=</span>output/nextclade.auspice.json <span class="dt">\</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>   <span class="at">--output-tree-nwk</span><span class="op">=</span>output/nextclade.tree.nwk <span class="dt">\</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>   <span class="at">--output-translations</span><span class="op">=</span>output/nextclade_CDS_{cds}.translation.fasta.zst <span class="dt">\</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>   <span class="co"># 多种输入文件</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>   <span class="ex">data/sars-cov-2/sequences.fasta</span> <span class="dt">\</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>   my_sequences1.fasta.gz <span class="dt">\</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>   my_sequences2.fasta.xz</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>输入序列格式：fasta, gz, xz, bz2, zst，可以同时进行多个fasta文件分析。</p>
</section></section><section id="nextclade-输入文件" class="level2"><h2 class="anchored" data-anchor-id="nextclade-输入文件">Nextclade 输入文件</h2>
<p>由两部分组成：</p>
<ul>
<li>查询序列，可以是多个文件，也可以是单个文件多条序列。</li>
<li>由参考序列，参考树，注释文件等组成的“dataset”，可以是路径，也可以是压缩文件。</li>
</ul>
<section id="查询序列" class="level3"><h3 class="anchored" data-anchor-id="查询序列">查询序列</h3>
<p>fasta 格式，支持除<code>U</code>外的<a href="https://www.bioinformatics.org/sms/iupac.html">IUPAC characters</a>，必须有“id”且以<code>&gt;</code>开头。</p>
</section><section id="dataset" class="level3"><h3 class="anchored" data-anchor-id="dataset">dataset</h3>
<p>数据集“dataset”是nextclade运行时，针对输入序列所需要的一些底层文件。它至少要包含参考序列文件：</p>
<ul>
<li>
<code>reference.fasta</code>：参考序</li>
</ul>
<p>其他分析用文件：</p>
<ul>
<li><p><code>genome_annotation.gff3</code>：参考序列的注释文件，<a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">GFF3</a>格式</p></li>
<li><p><code>tree.json</code>：参考树，Auspice JSON v2格式，可以使用 https://auspice.us/ 进行可视化</p></li>
<li><p><code>pathogen.json</code>：datasets 配置文件，类似身份证</p></li>
</ul>
<p>及说明文件：</p>
<ul>
<li><code>README.md</code></li>
<li><code>CHANGELOG.md</code></li>
</ul>
<p>测试数据：</p>
<ul>
<li><code>sequences.fasta</code></li>
</ul>
<p><a href="https://github.com/nextstrain/nextclade_data/tree/master/data">nextstrain/nextclade_data</a>是官方和社区维护的数据集。自己也可以创建数据集——<a href="https://github.com/nextstrain/nextclade_data/blob/master/docs/dataset-curation-guide.md">dataset 自制手册</a> 。</p>
<section id="datasets-的-name-和-version" class="level4"><h4 class="anchored" data-anchor-id="datasets-的-name-和-version">datasets 的 name 和 version</h4>
<p>name 由四级路径名称组成，例如<code>nextstrain/sars-cov-2/wuhan-hu-1/orfs</code>表示：由 Nextstrain 维护 的 <strong>SARS-CoV-2</strong> ，使用 <strong>Wuhan-Hu-1/2019 (MN908947)</strong> 毒株作为参考序列，并且这个序列带有 ORFs（开放阅读框），而不是其他成熟蛋白注释信息。</p>
<p>version 和软件的版本概念一致，用来确保结果的可重复性；它是一个数据释放时间戳<code>YYYY-MM-DDTHH:MM:SSZ</code>。Nextclade Web 和 CLI 都默认使用最新的版本；Web 使用URL来选择版本，CLI可以直接选择。</p>
<p>要保证结果能够复现，需要：</p>
<ul>
<li><p>保证 Nextclade CLI 版本一致</p></li>
<li><p>保证 datasets 版本一致</p></li>
</ul>
<p>查看可用的datasets：<code>nextclade dataset list --only-names</code></p>
<pre class="text"><code>nextstrain/sars-cov-2/wuhan-hu-1/orfs
nextstrain/sars-cov-2/wuhan-hu-1/proteins
nextstrain/sars-cov-2/BA.2.86
nextstrain/flu/h1n1pdm/ha/CY121680
nextstrain/flu/h1n1pdm/ha/MW626062
nextstrain/flu/h1n1pdm/na/MW626056
nextstrain/flu/h3n2/ha/CY163680
nextstrain/flu/h3n2/ha/EPI1857216
nextstrain/flu/h3n2/na/EPI1857215
nextstrain/flu/vic/ha/KX058884
nextstrain/flu/vic/na/CY073894
nextstrain/flu/yam/ha/JN993010
nextstrain/rsv/a/EPI_ISL_412866
nextstrain/rsv/b/EPI_ISL_1653999
nextstrain/mpox/all-clades
nextstrain/mpox/clade-i
nextstrain/mpox/clade-iib
nextstrain/mpox/lineage-b.1
nextstrain/orthoebolavirus/ebov
nextstrain/orthoebolavirus/sudv
nextstrain/flu/h3n2/pb1
nextstrain/flu/h3n2/np
nextstrain/flu/h1n1pdm/pa
nextstrain/flu/h3n2/ns
nextstrain/flu/h1n1pdm/mp
nextstrain/flu/h1n1pdm/np
nextstrain/flu/h1n1pdm/ns
nextstrain/flu/h3n2/mp
nextstrain/flu/h3n2/pa
nextstrain/flu/h1n1pdm/pb2
nextstrain/flu/h1n1pdm/pb1
nextstrain/flu/h3n2/pb2
nextstrain/measles/genome/WHO-2012
nextstrain/measles/N450/WHO-2012
nextstrain/dengue/all
nextstrain/yellow-fever/prM-E
nextstrain/hmpv/all-clades/NC_039199
nextstrain/flu/vic/pa
nextstrain/flu/vic/pb1
nextstrain/flu/vic/np
nextstrain/flu/vic/mp
nextstrain/flu/vic/pb2
nextstrain/flu/vic/ns
nextstrain/rubella/E1
nextstrain/herpes/vzv/NC_001348
nextstrain/mumps/sh
nextstrain/mumps/genome
nextstrain/rubella/genome
enpen/enterovirus/ev-d68
community/isuvdl/mazeller/prrsv1/orf5/yimim2025
community/isuvdl/mazeller/prrsv2/orf5/yimim2023
community/neherlab/hiv-1/hxb2
community/moncla-lab/iav-h5/ha/2.3.4.4
community/moncla-lab/iav-h5/ha/all-clades
community/moncla-lab/iav-h5/ha/2.3.2.1
community/v-gen-lab/dengue/denv1
community/v-gen-lab/dengue/denv2
community/v-gen-lab/dengue/denv3
community/v-gen-lab/dengue/denv4
community/genspectrum/marburg/HK1980/all-lineages
community/pathoplexus/cchfv/L
community/pathoplexus/cchfv/S
community/pathoplexus/cchfv/M
community/v-gen-lab/chikV/genotypes</code></pre>
<p>下载datasets：</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb6"><pre class="sourceCode sh code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nextclade</span> dataset get <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>   <span class="at">--name</span> <span class="st">'nextstrain/sars-cov-2/wuhan-hu-1'</span> <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">--tag</span> <span class="st">'2021-06-25T00:00:00Z'</span> <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">--output-dir</span> <span class="st">'data/sars-cov-2'</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</section><section id="reference-sequence" class="level4"><h4 class="anchored" data-anchor-id="reference-sequence">Reference sequence</h4>
<p><code>--input-ref</code>/<code>-r</code></p>
<p>当参考序列是一个注释完善、广泛使用的高质量基因组（例如来自 RefSeq）时，能获得最佳结果，该基因组最好是完整且明确的（涵盖整个基因组，没有模糊的核苷酸）。</p>
<p>参考序列最好不要有“gap”<code>-</code></p>
</section><section id="genome-annotation" class="level4"><h4 class="anchored" data-anchor-id="genome-annotation">Genome annotation</h4>
<p><code>--input-annotation</code>/<code>-m</code></p>
<p><a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">GFF3</a> 格式，但有额外要求：</p>
<ul>
<li>gene 名唯一</li>
<li>CDS 名唯一</li>
<li>protein 名唯一</li>
</ul>
<pre class="text"><code>##gff-version 3
##sequence-region   . 266 29533
.   .   gene    26245   26472   .   +   .   gene_name=E
.   .   gene    26523   27191   .   +   .   gene_name=M
.   .   gene    28274   29533   .   +   .   gene_name=N
.   .   gene    266 13468   .   +   .   gene_name=ORF1a
.   .   gene    13468   21555   .   +   .   gene_name=ORF1b
.   .   gene    25393   26220   .   +   .   gene_name=ORF3a
.   .   gene    27202   27387   .   +   .   gene_name=ORF6
.   .   gene    27394   27759   .   +   .   gene_name=ORF7a
.   .   gene    27756   27887   .   +   .   gene_name=ORF7b
.   .   gene    27894   28259   .   +   .   gene_name=ORF8
.   .   gene    28284   28577   .   +   .   gene_name=ORF9b
.   .   gene    21563   25384   .   +   .   gene_name=S</code></pre>
<p>注释文件与后续的密码子比对，CDS翻译分析有关，如果不提供该文件，则不会生成肽段序列文件和检测氨基酸突变。</p>
</section><section id="reference-tree" class="level4"><h4 class="anchored" data-anchor-id="reference-tree">Reference tree</h4>
<p><code>--input-tree</code>/<code>-a</code></p>
<p>格式： Auspice JSON v2</p>
<p>由 <code>augur export</code> 生产，使用 <a href="https://auspice.us/">Auspice</a> 进行可视化</p>
</section><section id="pathogen-configuration" class="level4"><h4 class="anchored" data-anchor-id="pathogen-configuration">Pathogen configuration</h4>
<p><code>--input-pathogen-json</code>/<code>-R</code></p>
<p>使用 <code>nextclade schema write --for input-pathogen-json</code> 生成模板。</p>
<p>要求 <code>schemaVersion</code> 格式，其中 <code>files</code> 字段是必须的，</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb8"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">{</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="dt">"files"</span><span class="fu">:</span> <span class="fu">{</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"reference"</span><span class="fu">:</span> <span class="st">"reference.fasta"</span><span class="fu">,</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"pathogenJson"</span><span class="fu">:</span> <span class="st">"pathogen.json"</span><span class="fu">,</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"genomeAnnotation"</span><span class="fu">:</span> <span class="st">"genome_annotation.gff3"</span><span class="fu">,</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"treeJson"</span><span class="fu">:</span> <span class="st">"tree.json"</span><span class="fu">,</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"examples"</span><span class="fu">:</span> <span class="st">"sequences.fasta"</span><span class="fu">,</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"readme"</span><span class="fu">:</span> <span class="st">"README.md"</span><span class="fu">,</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"changelog"</span><span class="fu">:</span> <span class="st">"CHANGELOG.md"</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">}</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="fu">}</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>可选属性：</p>
<ul>
<li>
<code>attributes</code>：<code>name</code>,<code>refrence name</code>等信息</li>
<li>
<code>qc</code>：有该字段时执行QC检查，具体参数见 <a href="https://docs.nextstrain.org/projects/nextclade/en/stable/user/algorithm/06-quality-control.html">Algorithm: Quality control</a>。</li>
<li>
<code>compatibility</code>：数据集要求的 nextclade 最低版本</li>
<li>
<code>defaultCds</code>：</li>
<li>
<code>cdsOrderPreference</code>：</li>
<li>
<code>generalParams</code>：</li>
<li>
<code>alignmentParams</code>：</li>
<li>
<code>treeBuilderParams</code>：</li>
<li>
<code>phenotypeData</code>：</li>
<li>
<code>aaMotifs</code>：</li>
<li>
<code>mutLabels</code>：</li>
</ul></section></section></section><section id="nextclade-输出文件" class="level2"><h2 class="anchored" data-anchor-id="nextclade-输出文件">Nextclade 输出文件</h2>
<p>以 <code>nextclade run --input-dataset data/sars-cov-2 --output-all=output/ data/sars-cov-2/sequences.fasta</code> 的结果为例，<code>--output-all</code>参数表示输出所有结果（包括不同格式），<code>output/</code>表示输出路径。输出结果可以分为五大部分：</p>
<ul>
<li>Nucleotide alignment</li>
<li>Translations</li>
<li>Analysis results</li>
<li>Phylogenetic tree</li>
<li>Genome annotation of query sequences</li>
</ul>
<pre class="text"><code>nextclade.aligned.fasta

nextclade.cds_translation.ORF3a.fasta
nextclade.cds_translation.ORF6.fasta
nextclade.cds_translation.E.fasta
nextclade.cds_translation.ORF7a.fasta
nextclade.cds_translation.M.fasta
nextclade.cds_translation.ORF7b.fasta
nextclade.cds_translation.N.fasta
nextclade.cds_translation.ORF8.fasta
nextclade.cds_translation.ORF1a.fasta
nextclade.cds_translation.ORF9b.fasta
nextclade.cds_translation.ORF1b.fasta
nextclade.cds_translation.S.fasta

nextclade.tsv
nextclade.csv
nextclade.json
nextclade.ndjson

nextclade.auspice.json
nextclade.nwk

nextclade.gff
nextclade.tbl</code></pre>
<section id="nucleotide-alignment" class="level3"><h3 class="anchored" data-anchor-id="nucleotide-alignment">Nucleotide alignment</h3>
<p><code>nextclade.aligned.fasta</code></p>
<p><code>--output-fasta</code>/<code>-o</code> <code>&lt;FILENAME&gt;</code></p>
<p>由 <code>sequence alignment</code> 步骤生产，输出结果是 相较于参考序列的 fasta 格式文件，任何相较于参考序列的插入片段都被移除，这些片段可以在后面的 tabular 和 json 文件中找到。</p>
</section><section id="translations" class="level3"><h3 class="anchored" data-anchor-id="translations">Translations</h3>
<p>模板字符串：<code>nextclade.cds_translation.{cds}.fasta</code></p>
<p><code>--output-translations</code>/<code>-P</code> <code>&lt;TEMPLATE_STRING&gt;</code></p>
<p>由 <code>translation and peptide alignment</code> 步骤生成，每个 CDS/gene 一个结果，格式同上。</p>
</section><section id="analysis-results" class="level3"><h3 class="anchored" data-anchor-id="analysis-results">Analysis results</h3>
<p>突变检测、进化分支分配、质量控制和 PCR 引物变化的结果有：表格格式（TSV、CSV）或 JSON 格式（经典 JSON 或 NDJSON）。</p>
<section id="tabular" class="level4"><h4 class="anchored" data-anchor-id="tabular">tabular</h4>
<p><code>nextclade.tsv</code> or <code>nextclade.csv</code></p>
<p><code>--output-tsv</code>/<code>-t</code> or <code>--output-csv</code>/<code>-c</code></p>
<p>结果是“1-base”，位置采用参考序列坐标（插入序列会被移除），坐标范围是闭区间的。</p>
<p>“seqName”列往往不唯一，需要通过“index”将输入与输出进行关联。</p>
<p>存在有关clade的额外列，例如示例结果中的“Nextclade_pango”，它们来自于参考树中的<code>meta.extensions.clade_node_attrs</code>。</p>
<p>推荐使用tsv格式，csv格式由于历史原因存在分隔符错乱的情况。</p>
</section><section id="json" class="level4"><h4 class="anchored" data-anchor-id="json">JSON</h4>
<p><code>nextclade.json</code> or <code>nextclade.ndjson</code></p>
<p><code>--output-json</code>/<code>-J</code> or <code>--output-ndjson</code>/<code>-N</code></p>
<p>JSON 结果会比tabular格式的结果多。NDJSON适合大型数据分析。</p>
<p>结果是“0-base”，位置采用参考序列坐标（插入序列会被移除），坐标范围左闭右开。</p>
</section></section><section id="phylogenetic-tree" class="level3"><h3 class="anchored" data-anchor-id="phylogenetic-tree">Phylogenetic tree</h3>
<p><code>nextclade.auspice.json</code> or <code>nextclade.nwk</code></p>
<p><code>--output-tree</code>/<code>-T</code> or <code>--output-tree-nwk</code></p>
<p>由 <code>phylogenetic placement</code> 步骤生成 “Auspice JSON v2” 或 “Newick”格式</p>
</section><section id="genome-annotation-of-query-sequences" class="level3"><h3 class="anchored" data-anchor-id="genome-annotation-of-query-sequences">Genome annotation of query sequences</h3>
<p>此项结果仍在实验中！！！</p>
<p><code>nextclade.gff</code> or <code>nextclade.tbl</code></p>
<p><code>--output-annotation-gff</code> or <code>--output-annotation-tbl</code></p>
<p>注释结果来源于输入的GFF文件，但是注释的坐标参考查询序列。</p>
<p>两种格式：Genbank的<a href="https://www.ncbi.nlm.nih.gov/genbank/feature_table/">TBL</a>格式，GFF3格式</p>
<p>GFF文件中有<code>seq_index</code>信息，用来绑定查询序列，因为查询序列的“name”信息不唯一。</p>
<p>添加<code>--retry-reverse-complement</code>会有额外的<code>is_reverse_complement</code>列。</p>
<p>如果输入的GFF文件中CDS没有对应的gene，nextclade会在结果中创建虚拟的gene；同样的，也会创建虚拟的CDS。</p>
<section id="errors-and-warnings" class="level4"><h4 class="anchored" data-anchor-id="errors-and-warnings">Errors and warnings</h4>
<p>格外要注意在tabular或json格式的结果中有“警告”或“错误”的序列，它们在其他结果中不存在。</p>
</section></section></section><section id="algorithm" class="level2"><h2 class="anchored" data-anchor-id="algorithm">Algorithm</h2>
<p>nextclade 内部进行了并行化处理。</p>
<section id="sequence-alignment" class="level3"><h3 class="anchored" data-anchor-id="sequence-alignment">Sequence alignment</h3>
<p>采用的比对策略：“a banded local alignment algorithm with affine gap-cost”</p>
<p>可以使用 <code>--alignment-preset</code> 中的预制参数，也可以在 <code>pathogen.json</code> 中设置，或在 <code>run</code> 中设置，优先级依次提高。</p>
<p>正链失败会尝试负链，不建议小于100个核苷酸的序列进行分析，如果查询序列与参考序列差异过大可能也会失败。</p>
<p>序列比对通常会存在模糊匹配结果，如果提供了基因组注释，Nextclade会在密码子的起始处（由上述模式中的|字符分隔）使用较低的缺口开放罚分，从而在可能的情况下将缺口锁定在阅读框内。同样，在存在歧义的情况下，Nextclade会优先将缺口放置在基因外部。</p>
</section><section id="translation" class="level3"><h3 class="anchored" data-anchor-id="translation">Translation</h3>
<p>当dataset中提供GFF3注释文件时运行。注释文件中的CDS可以通过<code>--cds-selection</code>/<code>-g</code>来选择。</p>
<p>将查询序列的CDS翻译后与参考序列的进行比对，比对算法同上。</p>
</section><section id="phylogenetic-placement" class="level3"><h3 class="anchored" data-anchor-id="phylogenetic-placement">Phylogenetic placement</h3>
<p>计算查询序列相较于参考序列的突变与其他树节点的距离，找到距离最近的节点。距离相近的节点的优先级可以在参考树中进行设置。</p>
<p><span class="math display">\[
D = M_{ref} + M_{query} - 2 M_{agree} - M_{disagree} - M_{unknown}
\]</span></p>
<p>在查看最近距离节点的过程中，不会将以放置的查询序列纳入计算。只有在后续进行贪婪型树的重构时才会将查询序列纳入计算。</p>
<p>将查询序列与树最近节点分开的突变是私有突变，二者共有的突变是共有突变。QC质控会检查私有突变是否是由测序或组装错误引起，同时检查这些突变在基因组上的位置。</p>
<p>nextclade 在设计时，考虑到了普通PC上的浏览器性能，所以树的准确性依赖参考树的多样性（dataset/tree.json）。</p>
<p>输入的系统发育树的根必须与输入的参考（根）序列相对应。如果参考序列与根节点的序列不同，则必须将两者之间的差异作为根节点的祖先突变添加进去。当遇到树中的多样性与参考序列之间存在不一致时，Nextclade 会报错。</p>
</section><section id="clade-assignment" class="level3"><h3 class="anchored" data-anchor-id="clade-assignment">Clade assignment</h3>
<p>nextclade 在参考树中使用 nextstrain 定义的“clade”和 PANGO 定义的 “lineage” 两种谱系分类方式。</p>
<p>分配时，基于 Phylogenetic placement 步骤中的结果；只能分配树中已有的 clade/lineage，无法识别参考树里没有的，它只会就近分配。</p>
</section><section id="mutation-calling" class="level3"><h3 class="anchored" data-anchor-id="mutation-calling">Mutation calling</h3>
<p>nextclade 会统计三种突变类型：替换，缺失，插入；同时统计<code>N</code>、非<code>ATCGN</code>等信息。氨基酸突变检测需要提供注释文件。</p>
<p>对于查询序列相较于树节点的私有突变，nextclade会进一步分析，将其分为：</p>
<ul>
<li>回复突变：极大可能是测序时导致</li>
<li>标记突变：可能是测序时导致</li>
<li>未标突变：不太可能是测试时导致</li>
</ul>
<p>三种突变在分支过程中的权重不同。</p>
</section><section id="quality-control-qc" class="level3"><h3 class="anchored" data-anchor-id="quality-control-qc">Quality Control (QC)</h3>
<p>在 dataset 中的 <code>pathogen.json</code> 中可以设置质控标准，也即每种病毒的质控标准不同。</p>
<p>质控的检测指标有：</p>
<ul>
<li>Missing data (N)</li>
<li>Mised sites (M)</li>
<li>Private mutations (P)</li>
<li>Mutation clusters (C)</li>
<li>Stop codons (S)</li>
<li>Frame shifts (F)</li>
</ul>
<p>QC 评分是多种指标的综合结果:</p>
<p><span class="math display">\[
S = \sum_{i} \frac{S_i^2}{100}
\]</span></p>
<p>QC质控结果只是参考，并不代表真实的质量情况。</p>
</section><section id="detection-of-pcr-primer-changes" class="level3"><h3 class="anchored" data-anchor-id="detection-of-pcr-primer-changes">Detection of PCR primer changes</h3>
<p>当 dataset 中的 <code>pathogen.json</code> 中配置了PCR引物表时，触发运行。</p>


</section></section><a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><script src="https://utteranc.es/client.js" repo="hanguojun007/Blog" issue-term="pathname" theme="github-light" crossorigin="anonymous" async="">
</script>
</div> <!-- /content -->



</body></html>